legend-engine-xt-openapi-generation/src/main/java/org/finos/legend/engine/generation/OpenApiArtifactGenerationExtension.java

```diff
@@ -0,0 +1,56 @@
+//  Copyright 2022 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.generation;
+
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.Artifact;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.ArtifactGenerationExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.pure.generated.Root_meta_external_format_openapi_metamodel_Server_Impl;
+import org.finos.legend.pure.generated.Root_meta_legend_service_metamodel_Service;
+import org.finos.legend.pure.generated.core_external_format_openapi_generation;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.finos.legend.engine.shared.core.operational.Assert.assertTrue;
+
+public class OpenApiArtifactGenerationExtension implements ArtifactGenerationExtension
+{
+
+    private static final String ROOT_PATH = "openapi_spec";
+
+    @Override
+    public String getKey()
+    {
+        return ROOT_PATH;
+    }
+
+    @Override
+    public boolean canGenerate(PackageableElement element)
+    {
+        return element instanceof Root_meta_legend_service_metamodel_Service;
+    }
+
+    @Override
+    public List<Artifact> generate(PackageableElement element, PureModel pureModel, PureModelContextData data, String clientVersion)
+    {
+        assertTrue(this.canGenerate(element), () -> "OpenApi extension only supports service elements");
```

Comment:
The platform is already calling `canGenerate` before calling generate, hence this check is not necessary,

Reply:
Thanks for letting me know, I was not aware platform calls it already.